airgun/views/redhat_repository.py

```diff
@@ -0,0 +1,271 @@
+from wait_for import wait_for
+from widgetastic.widget import (
+    GenericLocatorWidget,
+    Text,
+    TextInput,
+    View,
+)
+
+from airgun.views.common import BaseLoggedInView
+from airgun.widgets import (
+    ActionsDropdown,
+)
+
+SPINNER_LOCATOR = ".//div[contains(@class, 'spinner')]"
+
+
+def _wait_for_spinner(widget):
+    """Wait for any spinner to disappear from widget"""
+    wait_for(
+        lambda: widget.is_displayed and not widget.browser.elements(
+            SPINNER_LOCATOR, parent=widget),
+        timeout=60,
+        delay=1,
+        logger=widget.logger
+    )
+
+
+class AvailableRepositoryItem(GenericLocatorWidget):
+    """The widget representation of Available repository item of an Available repository set."""
+    ENABLE_BUTTON = './/button'
+    TEXT = './/span'
+
+    @property
+    def text(self):
+        """Return the text representation of this repository eg: architecture + OS version if
+        available.
+        """
+        return self.browser.text(self.TEXT, parent=self)
+
+    def read(self):
+        """Read this widget by return it's text representation."""
+        return self.text
+
+    def enable(self):
+        """Enable this repository."""
+        self.browser.click(self.ENABLE_BUTTON, parent=self)
+        _wait_for_spinner(self.parent)
+
+    def fill(self, value):
+        """Fill this widget by enabling the repository if value is True"""
+        if value is True:
```

Comment:
maybe we should rise some exception if value is smth else?

Reply:
I think we should not be explicit here. if True enable , if not ignore, that's what I think, this does not worth additional work from the user.